dom/events/scrolling/scrollend-event-fired-after-scroll.html

```diff
@@ -0,0 +1,88 @@
+<!DOCTYPE html>
+<meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1">
+<meta name="variant" content="?include=programmatic-scroll"/>
+<meta name="variant" content="?include=wheel-scroll"/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="/common/subset-tests-by-key.js"></script>
+<script src="scroll_support.js"></script>
+<style>
+#scroller {
+  width: 500px;
+  height: 500px;
+  overflow: scroll;
+  border: solid black 5px;
+}
+#spacer {
+  width: 5000px;
+  height: 5000px;
+}
+</style>
+
+<body style="margin:0" onload=runTests()>
+  <div id="scroller">
+    <div id="spacer"></div>
+  </div>
+</body>
+
+<script>
+
+let scroller = document.getElementById("scroller");
+let scroll_arrived_after_scroll_end = false;
+let scroll_end_event_count = 0;
+let scroll_event_count = 0;
+
+scroller.addEventListener("scroll", () => {
+  scroll_event_count += 1;
+  if (scroll_end_event_count != 0) {
+    scroll_arrived_after_scroll_end = true;
+  }
+});
+
+scroller.addEventListener("scrollend", () => {
+  scroll_end_event_count += 1
+});
+
+function runTests() {
+  async function wheelScroll() {
+    await new test_driver.Actions()
+      .addWheel("wheel1")
+      .scroll(50, 50, 0, 100, {origin: scroller})
+      .send();
+  }
+
+  async function programmaticScroll() {
+    spacer.scrollIntoView({behavior: "smooth", block: "end", inline: "end"});
+  }
+
+  async function runTestForScrollFn(runScroll, t) {
+    await waitForCompositorReady();
+    await waitForCompositorCommit();
+
+    assert_equals(scroller.scrollTop, 0, "Scroller has not scrolled");
+
+    await runScroll();
+
+    await waitFor(() => { return scroll_end_event_count != 0; });
+
+    assert_true(scroll_event_count > 0, "A scroll event has fired");
+
+    await waitForCompositorCommit();
```

Comment:
I'd prefer using two rAFs instead of `waitForCompositorCommit`, the function might be changed in future.